psqlgraph/psql.py

```diff
@@ -184,6 +194,12 @@ def session_scope(self, session=None, can_inherit=True,
         else:
             local = self.current_session()
 
+        # apply auto_flush and read only settings
+        if not inherited_session:
+            local.autoflush = auto_flush
+        if read_only:
+            local.execute("SET TRANSACTION READ ONLY")
```

Comment:
do we need to worry about `inherited_session` for the `read_only` flag as well?

Reply:
Yes, it will end up running this multiple times which is not intended, so makes sense to only do it for non inherited sessions